cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/ConnectionUtils.java

```diff
@@ -138,13 +138,17 @@ public static ConnectorSpec filterSpecWithPluginNameAndType(ConnectorSpec spec,
     ConnectorSpec.Builder specBuilder = ConnectorSpec.builder();
     specBuilder.setSchema(spec.getSchema());
     for (PluginSpec pluginSpec : spec.getRelatedPlugins()) {
-      if (Strings.isNullOrEmpty(pluginName) && pluginType.equalsIgnoreCase(pluginSpec.getType())) {
-        specBuilder.addRelatedPlugin(pluginSpec);
+      if (Strings.isNullOrEmpty(pluginName)) {
+        if (pluginType.equalsIgnoreCase(pluginSpec.getType())) {
```

Comment:
They can be both null but then it will not try to filter and return the old spec, https://github.com/cdapio/cdap/pull/14035/files#diff-83f5b42c5f5e49154ee43311ff93f5bedcc6b08dd35a9d5faacf4d62e067718fR135

Reply:
got it , thanks. missed it as it was hidden